cmake/modules/BuildRocksDB.cmake

```diff
@@ -75,10 +78,12 @@ function(build_rocksdb)
     set(make_cmd ${CMAKE_COMMAND} --build <BINARY_DIR> --target rocksdb)
   endif()
 
+  find_program(PATCH_EXECUTABLE patch)
   ExternalProject_Add(rocksdb_ext
     SOURCE_DIR "${rocksdb_SOURCE_DIR}"
     CMAKE_ARGS ${rocksdb_CMAKE_ARGS}
     BINARY_DIR "${rocksdb_BINARY_DIR}"
+    PATCH_COMMAND ${PATCH_EXECUTABLE} -p1 -i ${_build_rocksdb_list_dir}/rocksdb-gcc-13.patch
```

Comment:
Do we need to patch rocksdb on our own? How about merging those changes with the rockdb's upsteam and moving forward the `ceph/rocksdb` fork?

Reply:
agreed, we shouldn't need to patch our own submodule; we could start by merging this change to the ceph/rocksdb fork and carry it there until it merges upstream

Reply:
Right, good point. The patch was because I actually started working on fixing gcc 13 issues in SUSE's downstream pacific version, and I wasn't sure I wanted to touch the rocksdb submodule in that context.

Reply:
Submitted in https://github.com/ceph/rocksdb/pull/44